### DISCUSSION TITLE
Loading hierarchical translation files

### DISCUSSION THREAD

**Discussion by MMRandy** [i18next/i18next#1526] [General]
Thanks for such a well thought out, comprehensive library!  In fact, the breadth of the offering is the reason for my question.  I have a need to load a root level language translation file out of the gate, but then merge in keys that might be overridden in a  brand file (retailer may have multiple brands, each of which could override each default lang key value value).

Given all the options with plugins, namespaces, contexts, postprocessors....what would be the best way to accomplish this so that I can pass in the brand to the translate (t) function in my React components and get back the proper value if it is indeed overridden by a brand key for that language?  Just not sure if I should work on a robust merge right up front in the init to compose a large fiel with all possible translations and then use the context for the brand?  Or start with just the language file and use a postprocessor to look up the brand in a separate file?

Thanks for any guidance here and for all the work you've done on this lib!

**Comment by jamuhl** [comment]
One option is using "custom languages" those are defined like `en-US-x-customer1` -> so if you got control over the language set this can be an option.

Another option is creating an own backend: https://www.i18next.com/misc/creating-own-plugins#backend not to difficult and you can easily override what should be loaded.

Last option is using function on resource store: https://www.i18next.com/overview/api#resource-handling

**Reply by MMRandy** [reply]
Going the custom backend was the right call..does exactly what I need it to do. Thanks!
